Feature: Add support for running migration in batches (#7755)
impress-org/givewp: src/Framework/Migrations/readme.md

Reviewer: We don't necessarily need to do it in this PR, but I'd like to add a "Best Practices" section based on our conversations. It's clear that the best built migration system can still be fragile if good practices aren't followed.

Also, let's document the various migration log statuses and what they indicate.

# Migrations

We can think of migrations like version control for database.
We use them to create new tables, edit existing tables, and data manipulation.
Migrations are basically a representation of db schema at some point in time and that's why it's __important to never use models inside migration__.

## Migration types

There are two types of migrations supported in Give core - __Standard__ and __Batch__ migrations.

### Standard migration
We use these migrations each time we want to:
- Create new tables
- Edit existing tables (add/remove/update table columns)
- Delete tables

### Batch migration
Batch migrations, as name suggests, are processed in batches. We use them __only when we need to process large data sets__.

> __Important__: You will never use this type of migration for table creation, or table editing.

Reviewer: ```suggestion
__Important__: Do not use this type of migration for table creation or table editing.
```
